src/app/page.tsx

```diff
@@ -26,25 +26,48 @@ const fetchDataInBatches = async (count: number): Promise<{ batch: HomepageEntry
   const folders = ["lightning-dev", "bitcoin-dev"];
   let result: HomepageEntryData[] = [];
 
+  const currentDate = new Date();
+
+  const startYear = currentDate.getFullYear();
+  const endYear = 2010;
+  const startMonth = currentDate.getMonth().toString();
+  const currentMonth = monthsInOrder[startMonth];
+
+  const allpossibleMonths = createMonthsFromKeys(startYear, endYear);
+  const sliceIndex = allpossibleMonths.findIndex((month) => month.slice(0, -5) === currentMonth);
+
+  let monthsToBegin = allpossibleMonths.slice(sliceIndex);
+
+  const LTDIRECTORY = `${process.cwd()}/public/static/static/${folders[0]}/${monthsToBegin[count]}`;
+  const BTDIRECTORY = `${process.cwd()}/public/static/static/${folders[1]}/${monthsToBegin[count]}`;
+
+  const isFolderPresent = fs.existsSync(LTDIRECTORY) || fs.existsSync(BTDIRECTORY);
+
+  if (!isFolderPresent) {
+    monthsToBegin = allpossibleMonths.slice(sliceIndex + 1);
```

Comment:
does this account for if subsequent month isn't present as well?

it's not recursive and there is no guarantee that the previous month is available
this check should run till it finds a folder present in either directory

Apart from that, beautiful implementation. 
You went with the current month from date implementation for sorting 😌, it should be more efficient 👍

Reply:
So this check is for ensuring the beginning point we're starting from has files we can read. After generating all the possible months, the starting point is set to the current month you're on example October. 

We may be at the beginning of the month and not have any data to show or the folders from either directory are not present yet. For example, if I was using tldr on the 1st and 2nd Oct, the All activity section will turn up blank because there are no folders for Oct_2023 but we are already in that current month and the function starts reading from the current month. 

So what this does is it checks if the month you're starting has any data in both directories and if it doesn't it starts from the next month which would certainly have files there, and makes that month the starting point.

Reply:
okay okay, I get that. What i'm saying is that there's no guarantee the previous month would certainly have files.

Like a longterm approach. maybe it's sometime in the future or past. assume `oct_2014` and `oct_2045`
The assumption that `sept_2014` or `sept_2045` are guaranteed to have files is not absolute (some month folders may be missing, some months may not have transcripts, the scrapper may fail for some month etc) or other bizzare circumstance so you can make the code robust to accomodate any bizzareness

So i'm saying a more exhaustive check maybe keep checking till you find a folder that has files, i.e `aug_2014` and `aug_2045` and it keeps going. Very slim chance this might happen but it just future proof that aspect in the code.

I hope this make sense?

Reply:
Oh, I understand now. It sure does make sense, let me look into that.

Reply:
awesome! Although there's no need delaying this PR for that addition you can add it in another PR
I'm approving this PR
nice work!